### DISCUSSION TITLE
Some queries on liionpack

### DISCUSSION THREAD

**Discussion by ihlas123** [pybamm-team/liionpack#295] [General]
**1. Charging Termination Condition:** Are there any ways to terminate the experimental condition? It seems that specifying the experimental protocol as "Charge at 5 A for 47 minutes" is necessary for the successful completion of the experiment (Please refer to the attached document). When I set the charging condition to greater than 47 minutes, the solver issues an error indicating that the high-voltage limit has been reached. If not understood, kindly refer #290 in issue section. 

[Charge termination condition.zip](https://github.com/pybamm-team/liionpack/files/15148262/Charge.termination.condition.zip)

**2. Discharge Capacity:** Concerning the doubt regarding discharge capacity, I have attached the document illustrating the issue. The solver generates an error stating "<class 'pybamm.expression_tree.unary_operators.ExplicitTimeIntegral'> does not implement _unary_evaluate". How can I obtain the discharge capacity vs time curve?

[Issue on discharge capacity.pdf](https://github.com/pybamm-team/liionpack/files/15148234/Issue.on.discharge.capacity.pdf)

**3. Effect of i0:** Another doubt I have is regarding the effect of i0. I configured a circuit with Np = 1, Ns = 1, V = 4.2, and I = 0, utilizing "Chen2020" as the parameter set. Is there any tangible effect of I (denoted as i0 in the circuit diagram)? When I set i0 = 0 or i0 = 5, the experiment runs successfully. However, when I set i0 greater than 5, the solver fails to execute completely, displaying "Low-voltage limit reached". My concern is that even if I specify i0, it does not seem to affect the battery. For instance, setting i0 = 10 A with a 1p1s configuration using Chen2020 results in successful discharge at 5 A for 1 hour, while exceeding 5 A for 1 hour leads to "Low-voltage limit reached". It appears that i0 merely represents the total current passing through the circuit without impacting its behavior.

[EFFECT OF I0.zip](https://github.com/pybamm-team/liionpack/files/15148263/EFFECT.OF.I0.zip)

**Comment by TomTranter** [comment]
@ihlas123 I have now pushed an update to the voltage limits. You should be able to calculate Discharge capacity by coulomb counting the output as I think this will not work. I0 has no effect on the solve and is replaced by the solver with the experiment currents

**Comment by ihlas123** [comment]
Hii,
Thank you for your response. I'm curious to know in which version of
Liionpack you implemented the termination condition. I attempted to use the
termination condition in the latest version of Liionpack (0.3.9), but
unfortunately encountered the same issue.

Looking forward to your reply.

Best regards,
Ihlas Ahamed.

**Reply by TomTranter** [reply]
For now you can get it by installing from source on the develop branch. I will push another official release quite soon

**Comment by ihlas123** [comment]
Hello,

Regarding our previous conversation, I'm having trouble finding the method
to import the* "test_on_push" *condition. Additionally, since I'm new to
GitHub, I would greatly appreciate it if you could provide detailed
instructions on the installation process.
Thank you for your assistance.

Best regards,
Ihlas Ahamed.

**Comment by TomTranter** [comment]
I pushed a new release to pip. test_on_push sounds like a github action. For developer install you need to clone the repo. Make an environment using venv or conda then cd to the directory you cloned locally and do
`pip install -e .`
